packages/javascript/bh-shared-ui/src/components/Navigation/MainNav.tsx

```diff
@@ -35,16 +38,18 @@ const MainNavLogoTextImage: FC<{
     );
 };
 
-const MainNavListItem: FC<{ children: ReactNode; route?: string; hoverActive: boolean }> = ({
+const MainNavListItem: FC<{ children: ReactNode; route?: string; hoverActive: boolean; onClick?: () => void }> = ({
     children,
     route,
     hoverActive,
+    onClick = () => {},
 }) => {
     const location = useLocation();
     const isActiveRoute = route ? location.pathname.includes(route.replace(/\*/g, '')) : false;
 
     return (
         <li
+            onClick={typeof onClick === 'function' ? onClick : () => {}}
```

Comment:
Looks like you got your bases covered. You have a default noop if nothing is provided, and you have a function type specified if a value is provided.
```suggestion
            onClick={onClick}
```